### PR TITLE
[REVIEW] String concatenation fix in `DataFrame.rename`

### PR DESCRIPTION
**Pull request by galipremsagar**
Fixes: #3285

### CODE REVIEW

File: python/cudf/cudf/core/dataframe.py [hunk 1]
```diff
@@ -1845,7 +1845,7 @@ def rename(self, mapper=None, columns=None, copy=True, inplace=False):
             for key, col in self._data.items():
                 if key in mapper:
                     if mapper[key] in out.columns:
-                        out_column = mapper[key] + ("cudf_" + str(postfix),)
+                        out_column = mapper[key] + "cudf_" + str(postfix)
```
Comment (kkraus14) [inline python/cudf/cudf/core/dataframe.py]: What is the Pandas behavior here? To allow duplicate column names?

Comment (galipremsagar) [reply python/cudf/cudf/core/dataframe.py]: Yep, pandas allows duplicate column names.

Comment (kkraus14) [reply python/cudf/cudf/core/dataframe.py]: Got it. Would suggest we just do `mapper[key] + "_" + str(postfix)` then as `z` --> `zcudf_1` is super messy whereas `z_1` seems better.

Comment (galipremsagar) [reply python/cudf/cudf/core/dataframe.py]: Done 👍

File: python/cudf/cudf/core/dataframe.py [hunk 2]
```diff
@@ -1830,7 +1830,7 @@ def rename(self, mapper=None, columns=None, copy=True, inplace=False):
           * Not supporting: index, level
 
         Rename will not overwite column names. If a list with duplicates it
```
Comment (trxcllnt) [inline python/cudf/cudf/core/dataframe.py]:
```diff
- If a list with duplicates it
+ If a list with duplicates is
```

Comment (galipremsagar) [inline python/cudf/cudf/core/dataframe.py]:
```suggestion
        Rename will not overwite column names. If a list with duplicates is
```

**Pull request merged**